src/app/search/search-filters.ts

```diff
@@ -1183,6 +1188,38 @@ function searchFilters(
       hasLight(item: DimItem) {
         return item.primStat && statHashes.has(item.primStat.statHash);
       },
+      curated(item: D2Item) {
+        if (!item || item.season <= 3) {
+          return false;
+        }
+
+        // TODO: remove if there are no false positives, as this precludes maintaining a list for curatedNonMasterwork
+        // const masterWork = item.masterworkInfo && item.masterworkInfo.statValue === 10;
+        // const curatedNonMasterwork = [792755504, 3356526253, 2034817450].includes(item.hash); // Nightshade, Wishbringer, Distant Relation
+
+        const legendaryWeapon =
+          item.bucket && item.bucket.sort === 'Weapons' && item.tier.toLowerCase() === 'legendary';
+
+        let oneSocketPerPlug = true;
+
+        item.sockets &&
```

Comment:
could consider using [.every()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)
```javascript
        return legendaryWeapon && item.sockets && item.sockets.sockets.filter(
          // Remove Ornaments and Trackers
          (socket) =>
            socket &&
            socket.plug &&
            ![2947756142, 3940152116].includes(socket.plug.plugItem.plug.plugCategoryHash)
          )
          .every((socket) => socket && socket.plugOptions.length === 1);
```

Reply:
this is very nice